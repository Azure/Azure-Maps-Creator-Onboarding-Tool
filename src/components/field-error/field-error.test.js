import { render } from '@testing-library/react';

import FieldError from './field-error';

describe('Field Error', () => {
  it('should render nothing when text is not provided', () => {
    const view = render(<FieldError />);
    expect(view.container).toBeEmptyDOMElement();
  });

  it('should render nothing when text is empty', () => {
    const view = render(<FieldError text={''} />);
    expect(view.container).toBeEmptyDOMElement();
  });

  it('should render component when text is provided', () => {
    const view = render(<FieldError text={'Sweden has 267,570 islands, the most of any country in the world.'} />);
    expect(view.container).toMatchSnapshot();
  });
});