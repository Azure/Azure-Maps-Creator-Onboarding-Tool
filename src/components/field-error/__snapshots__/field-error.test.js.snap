// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field Error should render component when text is provided 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 0.813rem;
  color: #A80000;
}

.emotion-1 {
  margin-right: 0.25rem;
  font-size: 0.813rem;
}

<div>
  <span
    class="emotion-0"
  >
    <i
      aria-hidden="true"
      class="emotion-1 root-106"
      data-icon-name="StatusErrorFull"
    >
      ÓÆê
    </i>
    Sweden has 267,570 islands, the most of any country in the world.
  </span>
</div>
`;
